src/hash_table.py

```diff
@@ -0,0 +1,39 @@
+"""Hash table implementation."""
+
+
+class HashTable(object):
+    """Hash Table."""
+
+    def __init__(self):
+        """Initialize hash table."""
+        self._size = 1024
+        self._table = [None] * self._size
+
+    def get(self, key):
+        """Look item up in table."""
+        if not isinstance(key, str):
+            raise TypeError(key, type(key), "Key must be a string!")
+
+        hashed = self._hashing(key, self._size)
+
+        if self._table[hashed]:
+            for i in self._table[hashed]:
+                if i[0] == hashed:
+                    return i[1]
+
+        raise KeyError(key)
+
+    def set(self, key, val):
+        """Insert into table."""
+        hashed = self._hashing(key, self._size)
+        if self._table[hashed]:
+            self._table[hashed].append((hashed, val))
```

Comment:
Why are you appending `(hashed, val)`? Shouldn't that value be associated with the key, not the hash number? What if you have a collision between 3 keys and they all had a hash ID of 243? That bucket would be `[(243, val1), (243, val2), (243, val3)]` which isn't at all useful.